src/main/java/org/eclipse/xpanse/terraform/boot/terraform/TerraformToolHelper.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.terraform.boot.terraform;
+
+import jakarta.annotation.Resource;
+import java.io.File;
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.InvalidTerraformToolException;
+import org.eclipse.xpanse.terraform.boot.terraform.utils.SystemCmd;
+import org.eclipse.xpanse.terraform.boot.terraform.utils.SystemCmdResult;
+import org.springframework.stereotype.Component;
+
+/**
+ * Defines methods for handling the terraform tool with version.
+ */
+@Slf4j
+@Component
+public class TerraformToolHelper {
+
+    private static final String OS_NAME = System.getProperty("os.name").toLowerCase();
+    private static final String OS_ARCH = System.getProperty("os.arch").toLowerCase();
+
+    private static final Pattern TERRAFORM_VERSION_PATTERN =
+            Pattern.compile("^(=|>=|<=)([vV])?\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$");
+    private static final Pattern START_WITH_OPERATOR_PATTERN =
+            Pattern.compile("^(=|>=|<=)");
+    private static final Pattern END_WITH_VERSION_NUMBER_PATTERN =
+            Pattern.compile("^\\D*\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$");
+    private static final int MAX_VERSION_LENGTH = 20;
+
+    @Resource
+    private SystemCmd systemCmd;
+
+    /**
+     * Get Terraform version number from version string.
+     *
+     * @param version the version string
+     * @return the version number
+     */
+    public String getTerraformVersionNumber(String version) {
+        Matcher matcher = validateVersionFormat(version);
+        return matcher.group(0).replaceAll("^(=|>=|<=)", "");
+    }
+
+    private Matcher validateVersionFormat(String version) {
```

Comment:
we need the validation at the API level. Similar to what you added in OCL last week.